services/pong/src/service.js

```diff
@@ -116,4 +127,29 @@ export default class Service {
     // Update the ping details.
     this._pingStore.updateServicePing(service, online, ms);
   }
+
+  // Ping device and update how long the request takes.
+  async _pingDevice(device, host) {
+    const session = ping.createSession();
+    let online;
+    let ms;
+
+    try {
+      ms = await new Promise((resolve, reject) => {
+        session.pingHost(host, (err, _target, sent, rcvd) => {
+          if (err) reject(err);
+          else resolve(rcvd - sent);
+        });
+      });
```

Comment:
you can totally promisify this, I don't know why it didn't work last time

Reply:
Hm normally promises expect to have one return variable instead of two... but yeah I'd imagine it'd work here. It's fine for now though.